thread_setup_unix.c

```diff
@@ -29,7 +32,7 @@ static void thread_id(GO_CRYPTO_THREADID_PTR tid)
     // least once on any thread with associated error state. The thread-local
     // variable needs to be set to a non-NULL value so that the destructor will
     // be called when the thread exits. The actual value does not matter.
-    (void) pthread_setspecific(destructor_key, (void*)1);
+    (void) pthread_setspecific(destructor_key, &stub);
```

Comment:
I assume no backwards compatibility issue with older versions of gcc/glibc?

Reply:
`&stub` is a good old pointer, so nobody should complain about it. At least our CI is happy, which uses a glibc version that doesn't implement `access` modes.